xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/async/MentionsData.java

```diff
@@ -168,6 +170,23 @@ public MentionsData setWikiId(String wikiId)
         return this;
     }
 
+    /**
+     * Increase the attempts counter.
+     */
+    public void increaseAttempts()
+    {
+        this.attempts++;
+    }
+
+    /**
+     *
+     * @return true if the number of attempts for the current data has reached 10. 
+     */
+    public boolean isFailed()
```

Comment:
I would try to find a better name. See also my comment above (when you put back the task at the beginning of the queue). If you need to read the JavaDoc to understand what the method does then it means something is not right.

Reply:
Renamed `tooManyAttempts()`